### PR TITLE
Fix Dict, Tuple and List

### PR DESCRIPTION
**Pull request by skymandr**
Fixes so that we use lowercase types instead of `Dict`, `Tuple` and `List`, as well as a few `dict()` etc. initialisations.

I also started looking at the numerous instances where we use `dict(a=foo, b=bar)` syntax rather than `{"a": foo, "b": bar}`, but there is a butt-load of those, and in many instances it may be justified.

I'd be happy to revert the places where I did implement this, if we agree that we prefer it this way.

### CODE REVIEW

File: scanomatic/data_processing/strain_handler.py [hunk 1]
```diff
@@ -38,10 +38,10 @@ def loadCSV2Numpy(
             will be set to nan, else they will be zero of that data type
     """
 
-    def _putInData(d, pos, m):
+    def _putInData(d: dict[tuple[int, int], list], pos: SupportsInt, m):
```
Comment (local-minimum) [inline scanomatic/data_processing/strain_handler.py:41]:
```suggestion
    def _putInData(d: dict[tuple[int, int], list], pos: Sequence[SupportsInt], m):
```
or something right?

File: scanomatic/data_processing/strain_handler.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 from logging import Logger
-from typing import Any, List, Union, Tuple
+from typing import Any, SupportsInt, Union
```
Comment (local-minimum) [inline scanomatic/data_processing/strain_handler.py:2]: Sequence e deprecated, ska importeras från `collections.abc` enligt docs

File: scanomatic/data_processing/strain_selector.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from typing import List, Sequence, Tuple
+from typing import Sequence
```
Comment (local-minimum) [inline scanomatic/data_processing/strain_selector.py:1]:
```suggestion
from collections.abs import Sequence
```

File: scanomatic/io/image_loading.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import glob
 import os
 from logging import Logger
-from typing import Optional, Sequence, Tuple
+from typing import Optional, Sequence
```
Comment (local-minimum) [inline scanomatic/io/image_loading.py:4]: sequence from `collections.abs`

File: scanomatic/data_processing/calibration.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
 from collections import namedtuple
 from enum import Enum
 from logging import Logger
-from typing import Any, Callable, Dict, Union
+from typing import Any, Callable, Union
```
Comment (local-minimum) [inline scanomatic/data_processing/calibration.py:5]: callable from `collections.abc`

File: scanomatic/image_analysis/signal.py [hunk 6]
```diff
@@ -98,37 +105,42 @@ def extrapolate_edges(edges, frequency, signal_length):
...
             good_measures.append(d / 2.0)
     return np.mean(good_measures)
 
 
-def get_perfect_frequency2(best_measures, guess_frequency, tollerance=0.15):
-
+def get_perfect_frequency2(
+    best_measures: SpikesArray,
+    guess_frequency: float,
+    tolerance=0.15,
```
Comment (BengtRydberg) [inline scanomatic/image_analysis/signal.py]:
```suggestion
    tolerance: float = 0.15,
```

File: scanomatic/data_processing/norm.py [hunk 7]
```diff
@@ -1,5 +1,5 @@
 from enum import Enum
-from typing import Literal, Optional, Sequence, Tuple
+from typing import Literal, Optional, Sequence
```
Comment (local-minimum) [inline scanomatic/data_processing/norm.py:2]: iaf sequence from `collections.abc`

File: scanomatic/data_processing/phenotyper.py [hunk 8]
```diff
@@ -5,7 +5,7 @@
 from io import StringIO
 from itertools import chain, product
 from logging import Logger
-from typing import Any, Callable, List, Optional, Union, Tuple
+from typing import Any, Callable, Optional, Union
```
Comment (local-minimum) [inline scanomatic/data_processing/phenotyper.py:8]: callable from `collections.abc` iaf

File: scanomatic/data_processing/strain_handler.py (hunk 1 above)
Comment (local-minimum) [reply scanomatic/data_processing/strain_handler.py:41]: perhaps enough with `Iterable`.

File: scanomatic/data_processing/strain_handler.py (hunk 2 above)
Comment (skymandr) [reply scanomatic/data_processing/strain_handler.py:2]: Rätt sätt verkar vara att importera från collections direkt, inte från abc, tycker jag. Det funkar i alla fall.

**Review by local-minimum — APPROVED** [review]
lgtm

**Comment by skymandr** [comment]
For the record, before starting all this we had
```
Found 182 errors in 50 files (checked 162 source files)
```
Now we have
```
Found 171 errors in 38 files (checked 163 source files)
```
...

Not quite working backwards but still...

**Review by BengtRydberg — APPROVED** [review]
LGTM

**Pull request merged**